crates/wasmparser/src/validator/core.rs

```diff
@@ -256,7 +257,8 @@ impl ModuleState {
 
         let mut ops = expr.get_operators_reader();
         while !ops.eof() {
-            ops.visit_with_offset(&mut validator)??;
+            validator.offset = ops.original_position();
+            ops.visit_operator(&mut validator)??;
```

Comment:
Will explore how to remove the footgun of potentially forgetting to set the offset here too.

Reply:
I didn’t change anything here, but I realized this isn’t actually too bad as everything is scoped within this same function and so potential for misuse is minimal.